server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3213,6 +3214,13 @@ public void snapshotShard(SnapshotShardContext context) {
     }
 
     private void doSnapshotShard(SnapshotShardContext context) {
+        final long startTimeInMillis = threadPool.absoluteTimeInMillis();
+        blobStoreSnapshotMetrics.shardSnapshotStarted();
+        context.addListener(
+            ActionListener.running(
+                () -> blobStoreSnapshotMetrics.shardSnapshotCompleted(threadPool.absoluteTimeInMillis() - startTimeInMillis)
+            )
+        );
```

Comment:
We track start and finish shard here, which means we don't include the time in the queue, we could move this to include time in the queue as well?